File: 520.detect-capital.py
#
# @lc app=leetcode id=520 lang=python3
#
# [520] Detect Capital
#

# @lc code=start
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        return word.isupper() or word.islower() or word[0].isupper() and word[1:].islower()
# @lc code=end

File: 520.detect-capital.cs
/*
 * @lc app=leetcode id=520 lang=csharp
 *
 * [520] Detect Capital
 */

// @lc code=start
public class Solution {
    public bool DetectCapitalUse(string word) {
        return word.All(c => char.IsUpper(c)) || word.All(c => char.IsLower(c)) || (char.IsUpper(word[0]) && word.Substring(1).All(c => char.IsLower(c)));
    }
}
// @lc code=end

